### ISSUE TITLE
Increasing Forecasting Horizon without affecting training window in backtesting_forecaster function

### ISSUE THREAD

**Issue by udishadc** [JoaquinAmatRodrigo/skforecast#764]
I'm using the backtesting_forecaster function in skforecast for time series forecasting.

Currently, the `steps` parameter controls both the forecasting horizon and the number of steps by which training window is moving in each iteration. Is there a way to independently adjust the forecasting horizon without affecting the training window?

**Code Snippet:**
```python
steps = 24
metric, predictions_backtest = backtesting_forecaster(
                                   forecaster         = forecaster,
                                    y                  = data_train["y"][:24*100],                                
                                   initial_train_size = 24*10,
                                   fixed_train_size   = True,
                                   steps              = steps,
                                   metric             = 'mean_squared_error',
                                   refit              = True,
                                   verbose            = True,
                                   show_progress      = True
                               
)
print(f"Backtest metric (MSE): {metric}") 
```

Output:
```python
Information of backtesting process
----------------------------------
Number of observations used for initial training: 240
Number of observations used for backtesting: 2160
    Number of folds: 90
    Number of steps per fold: 24
    Number of steps to exclude from the end of each train set before test (gap): 0

Fold: 0
    Training:   2006-01-01 00:00:00 -- 2006-01-10 23:00:00  (n=240)
    Validation: 2006-01-11 00:00:00 -- 2006-01-11 23:00:00  (n=24)
Fold: 1
    Training:   2006-01-02 00:00:00 -- 2006-01-11 23:00:00  (n=240)
    Validation: 2006-01-12 00:00:00 -- 2006-01-12 23:00:00  (n=24)
Fold: 2
    Training:   2006-01-03 00:00:00 -- 2006-01-12 23:00:00  (n=240)
    Validation: 2006-01-13 00:00:00 -- 2006-01-13 23:00:00  (n=24)
Fold: 3
    Training:   2006-01-04 00:00:00 -- 2006-01-13 23:00:00  (n=240)
    Validation: 2006-01-14 00:00:00 -- 2006-01-14 23:00:00  (n=24)
Fold: 4
    Training:   2006-01-05 00:00:00 -- 2006-01-14 23:00:00  (n=240)
    Validation: 2006-01-15 00:00:00 -- 2006-01-15 23:00:00  (n=24)
Fold: 5
    Training:   2006-01-06 00:00:00 -- 2006-01-15 23:00:00  (n=240)
...
Fold: 89
    Training:   2006-03-31 00:00:00 -- 2006-04-09 23:00:00  (n=240)
    Validation: 2006-04-10 00:00:00 -- 2006-04-10 23:00:00  (n=24)
```

**Comment by JoaquinAmatRodrigo** [comment]
Hi @udishadc,
Thanks for using skforecast.
Currently, the training window is tied to the forecast horizon (`steps`).  Three different configurations are allowed:

+ `refit = False` : only the initial training is applied.
+ `refit = True` : the training window moves `steps` positions in each fold
+ `refit = <integer>` : the training is repeated each `<integer>` folds.

The reason for this behavior is that the current skforecast API only allows 1 predicted value per step.
What kind of behavior are you looking for? Your feedback may help us include new features in future releases.

**Comment by udishadc** [comment]
Hi @JoaquinAmatRodrigo,
Thank you for the clarification.
I was looking to evaluate model performance across different forecasting horizons while keeping the training window constant.
I understand the current limitations of the steps parameter and the reasons behind it.
Thank you.

**Comment by JoaquinAmatRodrigo** [comment]
Hi @lorenzomezzini,
Thanks for your interest in collaborating on Skforecast.

Do you have any ideas on how you would approach this?

**Comment by lorenzomezzini** [comment]
Currently, the size of the test set is strictly equal to the number of steps.
This makes it impossible to implement certain backtesting strategies for example: running a 60-day forecast every Monday.

My first idea is to introduce a new optional attribute to `TimeSeriesFold `, e.g.: `forward_steps` (by default is equal to `steps` for backward compatibility)

- `steps` would continue to define the length of the test set.

- `forward_steps` would control how much the training set moves forward at each iteration.

I had a quick look of the codebase and I belive that most of the logical changes should be in here (starting at line 920 of [_split.py](https://github.com/skforecast/skforecast/blob/master/skforecast/model_selection/_split.py)):
```python
  while self.initial_train_size + (i * self.forward_steps) + self.gap < len(index):
  
  	if self.refit:
  		# If `fixed_train_size` the train size doesn't increase but moves by
  		# `steps` positions in each iteration. If `False`, the train size
  		# increases by `steps` in each iteration.
  		train_iloc_start = i * (self.forward_steps) if self.fixed_train_size else 0
  		train_iloc_end = self.initial_train_size + i * (self.forward_steps)
  		test_iloc_start = train_iloc_end
  	else:
  		# The train size doesn't increase and doesn't move.
  		train_iloc_start = 0
  		train_iloc_end = self.initial_train_size
  		test_iloc_start = self.initial_train_size + i * (self.forward_steps)
  
  	if self.window_size is not None:
  		last_window_iloc_start = test_iloc_start - self.window_size
  	test_iloc_end = test_iloc_start + self.gap + self.steps
  
  	partitions = [
  		idx[train_iloc_start:train_iloc_end],
  		(
  			idx[last_window_iloc_start:test_iloc_start]
  			if self.window_size is not None
  			else []
  		),
  		idx[test_iloc_start:test_iloc_end],
  		idx[test_iloc_start + self.gap : test_iloc_end],
  	]
  	folds.append(partitions)
  	i += 1

```



This change should preserve backward compatibility and give users more control over the backtesting frequency.

**Comment by JoaquinAmatRodrigo** [comment]
This is a great starting point, nice work!
To make sure everything integrates smoothly, we’ll need to check compatibility with the backtesting functions.
I’m currently focused on the 0.17.0 release, but once that’s wrapped up, I’ll take a closer look at this 🙂

**Comment by lorenzomezzini** [comment]
Thanks! 

Sure,I'd love love to help after the 0.17.0 release if you want to move forward with this.

Regarding your question: I believe that the multiple prediction wouldn’t impact this specific issue, since the number of predictions per step remains the same. The purposed changes should affect only the date ranges, not how the predictions are generated or evaluated.

However, if you're referring to the more general case like "How would I handle backtesting metrics with multiple predictions?" that opens up a broader discussion. I'm definitely open to contributing ideas or code in that direction too, if needed!

**Comment by lorenzomezzini** [comment]
> Hi [@lorenzomezzini](https://github.com/lorenzomezzini), Picking up where we left off: I might be misunderstanding, but in your example of "running a 60-day forecast every Monday," wouldn’t that necessarily generate multiple predictions for some steps?
> 
> For instance, if you start on a Monday, you’d forecast days 1–60. The following Monday, you’d forecast days 8–68. That means days 8–60 would have two overlapping predictions. This overlap would repeat as many times as there are Mondays within the 60-day window.
> 
> Am I understanding your example correctly?

Hi @JoaquinAmatRodrigo,

I see what you mean now.

In our case, we haven’t yet looked into how SkForecast computes metrics during backtesting, since our focus was simila to running a 60-day forecast every Monday. In that setup, we were evaluating performance prediction-wise rather than day-wise.

So, following your example:
- The forecast for days 1–60 is used to compute the metric (MAE in our case) for the first Monday.
- The forecast for days 8–68 is used for the second Monday.
- And so on.

The final result is essentially a table showing “MAE over time”, one value per prediction run.
